lang/julia/README.md

```diff
@@ -0,0 +1,374 @@
+```
+$ mv julia.sif_latest.sif julia.sif
+```
+
+
+To run the container:
+
+```
+$ ./julia.sif 
+Hello world from: https://raw.githubusercontent.com/sylabs/examples/master/lang/julia/hello-world.jl
+For full tutorial, visit: https://github.com/sylabs/examples/lang/julia
+$
+```
+
+You can also do:
+
+```
+$ singularity run julia.sif
+```
+
+
+<br>
+
+
+### To build from a definition file:
+
+You will need root access to build from a recipe.
+
+
+First, make the definition file:
+
+```
+$ nano julia.def
+```
+```
+BootStrap: library
+From: ubuntu:16.04
+
+%runscript
+# run your script here, eg.
+# julia testing.jl
+echo 'println("hello world from julia script!!!")' | julia
+
+%environment
+export PATH=/julia-1.0.1/bin:$PATH
+export LD_LIBRARY_PATH=/julia-1.0.1/lib:/julia-1.0.1/lib/julia:$LD_LIBRARY_PATH
+export LC_ALL=C
+
+%post
+apt-get -y update
+# we are installing some basic packages,
+# you can install your own
+#apt-get -y install <YOUR_PACKAGE>
+
+# install some basic tools
+apt-get -y install curl tar gzip
+
+apt-get clean
+apt-get autoremove
+
+# now, download and install julia
+curl -sSL "https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.1-linux-x86_64.tar.gz" > julia.tgz
+tar -C / -zxf julia.tgz
+rm -f julia.tgz
+```
+
+Or, you can just download the definition file:
+
+```
+$ wget https://raw.githubusercontent.com/sylabs/examples/master/lang/julia/julia.def
+```
+
+Then you can modify the `julia.def` file as you need,
+e.g., installing other packages.
+
+
+
+Now, to build the container:
+
+```
+$ sudo singularity build julia.sif julia.def
+```
+
+<br>
+
+### To run a julia script:
+
+
+There are several ways of doing this.
+
+Jump to:
+ - [Run script by `shell`.](#running-script-by-shell)
+ - [Run script by `exec`.](#running-script-using-exec)
+ - [Run scritp by `%runscript`.](#embed-the-run-command-to-runscript)
+ - [Run script by embedding script to defintion file.](#embed-the-script-into-your-container)
+ - [Run script by pulling from web via `curl` or `wget`.](#run-the-script-by-pulling-from-the-web)
+
+
+<br>
+
+
+### Running script by shell:
+
+The test file:
+
+```
+$ nano testing.jl
+```
+```
+#!/usr/bin/env julia
+
+println("hello world from julia!")
+```
+
+Or just download the test julia file with `wget`:
+
+```
+$ wget https://raw.githubusercontent.com/sylabs/examples/master/lang/julia/hello-world.jl
+```
+
+
+Then shell into the container:
+
+```
+$ singulairty shell julia.sif
+```
+
+And then run the script:
+
+```
+> julia testing.jl
+```
+Or
+```
+> chmod +x testing.jl
+> ./testing.jl
+```
+
+<br>
+
+
+### Running script using `exec`:
+
+```
+$ singularity exec julia.sif julia testing.jl
+```
+
+<br>
+
+
+### Embed the run command to runscript:
+
+```
+$ nano julia.def
+```
+And add the command that will execute the script to: `%runscript`
+```
+BootStrap: library
+From: ubuntu:16.04
+
+%runscript
+# you can execute them by running:
+# singularity run julia.sif
+# run your script here, eg.
+julia testing.jl
```

Comment:
Change this to:
`julia "$@"` 

The run with:  `singularity run julia.sif testing.jl`